src/components/routes/qiCore/TestCaseRoutes.tsx

```diff
@@ -52,47 +53,37 @@ const TestCaseRoutes = () => {
       setLastMeasure(compareTo);
       setErrors(() => []);
       if (measure.cqlErrors || !measure.elmJson) {
-        setErrors((prevState) => [
-          ...prevState,
-          "An error exists with the measure CQL, please review the CQL Editor tab.",
-        ]);
+        localErrors.push(
+          "An error exists with the measure CQL, please review the CQL Editor tab."
+        );
       }
       if (!measure?.groups?.length) {
-        setErrors((prevState) => [
-          ...prevState,
-          "No Population Criteria is associated with this measure. Please review the Population Criteria tab.",
-        ]);
+        localErrors.push(
+          "No Population Criteria is associated with this measure. Please review the Population Criteria tab."
+        );
       }
-      if (!errors?.length) {
+
+      if (!localErrors.length) {
         measureService.current
           .fetchMeasureBundle(measure)
           .then((bundle: Bundle) => {
             setMeasureBundle(bundle);
           })
           .catch((err) => {
             setContextFailure(true);
-            setErrors((prevState) =>
-              prevState.length === 1 &&
-              prevState[0].includes(
-                "No Population Criteria is associated with this measure. Please review the Population Criteria tab"
-              )
-                ? [...prevState]
-                : [...prevState, err.message]
-            );
+            localErrors.push(err.message);
           });
       }
-      setErrors((prevState) => {
-        if (
-          measure?.errors?.includes(
-            MeasureErrorType.MISMATCH_CQL_POPULATION_RETURN_TYPES
-          )
-        ) {
-          return [...prevState, CQL_RETURN_TYPES_MISMATCH_ERROR];
-        }
-        return [
-          ...prevState.filter((s) => s !== CQL_RETURN_TYPES_MISMATCH_ERROR),
-        ];
-      });
+
+      if (
+        measure?.errors?.includes(
+          MeasureErrorType.MISMATCH_CQL_POPULATION_RETURN_TYPES
+        )
+      ) {
+        localErrors.push(CQL_RETURN_TYPES_MISMATCH_ERROR);
+      }
+      localErrors.filter((s) => s !== CQL_RETURN_TYPES_MISMATCH_ERROR);
```

Comment:
This has to be in else block.

Reply:
Yep, forgot to commit/push it

Reply:
Filter will not mutate the object, instead it returns a new array, so we need to assign the result it back to localErrors array

Reply:
Atleast that is what my understanding is, I was hoping there are test cases to prove me wrong.